Lib9c/Action/ClaimItems.cs

```diff
@@ -22,24 +25,37 @@ public class ClaimItems : GameAction, IClaimItems
         private const int MaxClaimDataCount = 100;
 
         public IReadOnlyList<(Address address, IReadOnlyList<FungibleAssetValue> fungibleAssetValues)> ClaimData { get; private set; }
+        public string Memo;
 
         public ClaimItems()
         {
         }
 
-        public ClaimItems(IReadOnlyList<(Address, IReadOnlyList<FungibleAssetValue>)> claimData)
+        public ClaimItems(IReadOnlyList<(Address, IReadOnlyList<FungibleAssetValue>)> claimData, string memo = null)
         {
             ClaimData = claimData;
+            Memo = memo;
         }
 
         protected override IImmutableDictionary<string, IValue> PlainValueInternal =>
-            ImmutableDictionary<string, IValue>.Empty
+            GetPlainValueInternal();
+
+        private IImmutableDictionary<string, IValue> GetPlainValueInternal()
+        {
+            var dict = ImmutableDictionary<string, IValue>.Empty
                 .Add(ClaimDataKey, ClaimData.Aggregate(List.Empty, (list, tuple) =>
                 {
                     var serializedFungibleAssetValues = tuple.fungibleAssetValues.Select(x => x.Serialize()).Serialize();
 
                     return list.Add(new List(tuple.address.Bencoded, serializedFungibleAssetValues));
                 }));
+            if (!string.IsNullOrEmpty(Memo))
+            {
+                dict = dict.Add(MemoKey, Memo.Serialize());
+            }
```

Comment:
If memo is an empty string, I think there is any reason to serialize it and include it in the chain. (I think network & space are wasted with unnecessary information.) Since the chain compares the serialized state, there are no problem.

Reply:
> If memo is an empty string, I think there is any reason to serialize it and include it in the chain. (I think network & space are wasted with unnecessary information.)

I think so. Then, I think it should restrict when the `plainValue[MemoKey] = ""` case as alternative.

Reply:
That's why I used `string.isNullorEmpty`, is there better suggestion?